### PR TITLE
[V1][core] Support allowed_token_ids in V1 sampler

### LINKED ISSUE #13058: [V1][Help Wanted] Porting missing sampling parameters to V1
### Anything you want to discuss about vllm.

To switch the engine from V0 to V1, we need to comprehensively support the sampling parameters in https://github.com/vllm-project/vllm/blob/main/vllm/sampling_params.py

While most of the key parameters are already supported, some of them are missing:

TODO (help wanted):
- [x] `n` (parallel sampling) #10980  @afeldman-nm 
- [x] `guided_decoding` (structured decoding) #12388  @aarnphm 
- [x] `logit_bias` #13079 @houseroad 
- [x] `min_p` #13191 @AoyuQC
- [ ] `bad_words` (originally implemented via logits processor) #13376 @22quinn 
- [x] `allowed_token_ids` (originally implemented via logits processor) #13210 @houseroad 

Parameters that will not be supported in V1:
* best_of
* logits_processors

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by catherinelee274**
Introduce allowed_token_id support in v1 Sampler.

Tested with:
pytest tests/v1/sample/test_sampler.py -k "test_sampler_allowed_token_ids"

Fixes #13058

### CODE REVIEW

**Comment by njhill** [comment]
Thanks @catherinelee274! There is actually a PR already open for this though: https://github.com/vllm-project/vllm/pull/13210

**Comment by WoosukKwon** [comment]
@catherinelee274 Closing this as #13210 is merged. Thanks!
